core/src/main/web/app/mainapp/components/notebook/codecell-directive.js

```diff
@@ -70,21 +70,24 @@
           if ($scope.isLocked()) {
             return false;
           }
-          if ($scope.cellmodel.input.hidden === true) {
-            return false;
-          }
-          return true;
+          return $scope.cellmodel.input.hidden !== true;
         };
 
         $scope.bkNotebook = getBkNotebookWidget();
+
         // ensure cm refreshes when 'unhide'
-        $scope.$watch('isShowInput()', function(newValue, oldValue) {
-          if ($scope.cm && newValue === true && newValue !== oldValue) {
-            bkUtils.fcall(function() {
-              $scope.cm.refresh();
-            });
+        var lastInputHidden = $(".code-cell-input").hasClass("ng-hide");
+        window.setInterval(function () {
```

Comment:
setInterval is outside of Angular scope. There is angular's $interval service that we should use for that https://docs.angularjs.org/api/ng/service/$interval

However, do we really want to poll every 10 miliseconds for every cell? So if we have 100 cells that's 100 intervals going off on every frame, 60 frames per second = 6000 function calls per second and every single one of them touches the DOM. That doesn't sound great.

Or am I misreading this code?